test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -299,6 +238,89 @@ def test_full_with_dtype_overridden(self):
         expected_tensor = torch.full((h, w), fill_value=fill_value, device=local_device, dtype=torch.double)
         self.assertEqual(expected_tensor, local_tensor)
 
+
+class TestShardParameter(ShardedTensorTestBase, MultiProcessTestCase):
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_shard_parameter(self):
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+            ],
+        )
+
+        fc = torch.nn.Linear(12, 12).cuda(self.rank)
+        shard_parameter(fc, 'weight', spec)
+
+        # Verify.
+        self.assertTrue(isinstance(fc.weight, _sharded_tensor.ShardedTensor))
+        local_shards = fc.weight.local_shards()
+        self.assertEqual(1, len(local_shards))
+        self.assertEqual(torch.Size([3, 12]), local_shards[0].tensor.size())
```

Comment:
nit: you probably need a sanity check to check the sharded local tensor is the same as a *shard* from original tensor.

original_fc = nn.Linear(12, 12)
fc = a deep copy of original_fc

for each rank -- check local_shard.tensor is the same as the corresponding part in original_fc.weight